### PR TITLE
✨ [NFT Details] Override NFT Class metadata with NFT metadata

### PR DESCRIPTION
**Pull request by nwingt**
Continue from #806

### CODE REVIEW

File: src/pages/nft/class/_classId/_nftId.vue [hunk 1]
```diff
@@ -318,31 +338,33 @@ export default {
...
-        store.dispatch('fetchNFTMetadata', classId),
+        store.dispatch('fetchNFTClassMetadata', classId),
         store.dispatch('lazyGetNFTPurchaseInfo', classId).catch(err => {
           if (err.response?.data !== 'NFT_CLASS_NOT_FOUND') {
             // eslint-disable-next-line no-console
             console.error(JSON.stringify(err));
           }
         }),
       ]);
+      nftMetadata = await store.dispatch('fetchNFTMetadata', {
```
Comment (williamchong) [inline src/pages/nft/class/_classId/_nftId.vue:364]: Not sure if we should/can put this into the above `Promise.all`

Comment (nwingt) [reply src/pages/nft/class/_classId/_nftId.vue:364]: The original action depends on NFT Class metadata for fallback, I think we can resolve that with computed properties.

File: src/store/modules/nft.js [hunk 2]
```diff
@@ -28,6 +29,9 @@ const mutations = {
   [TYPES.NFT_SET_NFT_CLASS_METADATA](state, { classId, metadata }) {
     Vue.set(state.metadataByClassIdMap, classId, metadata);
   },
+  [TYPES.NFT_SET_NFT_METADATA](state, { nftId, metadata }) {
+    Vue.set(state.metadataByNFTIdMap, nftId, metadata);
```
Comment (williamchong) [inline src/store/modules/nft.js:33]: class id +  nft id is the unique key, so we need to set class id too

File: src/store/modules/nft.js [hunk 3]
```diff
@@ -84,6 +88,7 @@ const getters = {
       (acc, val) => acc + val.length,
       0
     ),
+  getNFTMetadataById: state => id => state.metadataByNFTIdMap[id],
```
Comment (williamchong) [inline src/store/modules/nft.js:91]: class id + nft id is the unique key, so we need to pass class id too

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/store/modules/nft.js (hunk 3 above)
Comment (nwingt) [reply src/store/modules/nft.js]: Updated

**Pull request merged**